pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -347,6 +351,41 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 	return cluster, nil
 }
 
+func (p *Provisioner) setClusterStatusUpdating(cluster *v3.Cluster) {
+	v3.ClusterConditionUpdated.Unknown(cluster)
+	_, err := p.Clusters.Update(cluster)
+	if err != nil {
+		if !apierrors.IsConflict(err) {
+			logrus.Errorf("Failed to update cluster status [%s]: %v", cluster.Name, err)
+		} else {
+			backoff := wait.Backoff{
+				Duration: 100 * time.Millisecond,
+				Factor:   2,
+				Jitter:   0,
+				Steps:    4,
+			}
+
+			err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+				newCluster, err := p.Clusters.Get(cluster.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, nil
+				}
+				v3.ClusterConditionUpdated.Unknown(newCluster)
+				if _, err = p.Clusters.Update(newCluster); err != nil {
+					if apierrors.IsConflict(err) {
+						return false, nil
+					}
+					return false, fmt.Errorf("Failed to update cluster status [%s]: %v", cluster.Name, err)
+				}
+				return true, nil
+			})
+			if err != nil {
+				logrus.Errorf("%v", err)
```

Comment:
logging err with no additional contextual message is a bad pattern. Add some context. In this case of never getting true returned, you would get  a very generic WaitTimeoutExceeded error.
Also, should probably be logged at WARN level